nsxt/resource_nsxt_policy_tier1_gateway.go

```diff
@@ -397,6 +409,7 @@ func policyTier1GatewayResourceToInfraStruct(d *schema.ResourceData, connector *
 		RouteAdvertisementRules: routeAdvertisementRules,
 		Ipv6ProfilePaths:        ipv6ProfilePaths,
 		ResourceType:            &t1Type,
+		HaMode:                  &haMode,
```

Comment:
I think this should require NSX version check (specifying this value in early versions would likely lead to an error)

Reply:
For NSX < 4.0.0 this would cause ```Error:  Failed to create Tier1: Json de-serialization error: property ha_mode is unrecognized. (code 287)```